src/main/java/seedu/address/model/pet/Pet.java

```diff
@@ -0,0 +1,119 @@
+package seedu.address.model.pet;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+import seedu.address.model.tag.UniqueTagList;
+
+/**
+ * Represents a Pet in the applications.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Pet {
+
+    private final PetName petName;
+    private final PetAge petAge;
+    private final Species species;
+    private final PetGender petGender;
+    private final Client petOwner;
+    private final Appointment appointment;
```

Comment:
Instead of using the Client and Appointment classes (since they aren't created on your side and will hence cause the checks to fail), should you just use a dummy variable first?

Reply:
Resolved